File: AspNetCore_gdr2333_Front+Back/Back/Types/DataBase/PrivateMessage.fs
namespace Back.Types.DataBase

open System
open System.ComponentModel.DataAnnotations

type PrivateMessage(senderId: int64, receiverId: int64, content: string) =
    class
        [<Key>]
        member val PrivateMessageId = 0L with get

        member val SenderId = senderId with get
        member val ReceiverId = receiverId with get
        member val Content = content with get
        member val SendTime = DateTime.Now with get
        member val Checked = false with get, set
        member val CheckSucess = false with get, set
    end

File: AspNetCore_gdr2333_Front+Back/Back/Types/DataBase/PrivateMessage.cs
using System.ComponentModel.DataAnnotations;

namespace Back.Types.DataBase;

public class PrivateMessage
{
    [Key]
    public long PrivateMessageId { get; set; }

    public long SenderId { get; set; }

    public long ReceiverId { get; set; }

    public string Content { get; set; }

    public DateTime SendTime { get; set; }

    public bool Checked { get; set; }

    public bool CheckSuccess { get; set; }

    public bool Readed { get; set; }

    public PrivateMessage(long senderId, long receiverId, string content)
    {
        SenderId = senderId;
        ReceiverId = receiverId;
        Content = content;
        SendTime = DateTime.Now;
        Checked = false;
        CheckSuccess = false;
        Readed = false;
    }

    public PrivateMessage()
    {
    }
}
